### DISCUSSION TITLE
Can `upload sources` delete files from crowdin that are no longer in the repo?

### DISCUSSION THREAD

**Discussion by CraigSiemens** [crowdin/github-action#171] [Q&A]
Recently I was looking through our project in crowdin a saw a bunch of files that no longer exist in our repo. They've either been removed or renamed so the file at the original path no longer exists. I'm wondering if there's a way for the action to automatically remove files from the crowdin project that no longer exist in our repo.

I looked at [action.yml](https://github.com/crowdin/github-action/blob/master/action.yml) but I'm not seeing an option there for removing/deleteing files.

I did see the `upload_sources_args` option, so looking at the CLI I found `--delete-obsolete` which *might* do it, but I'm unsure whether an "obsolete file" is one that's been removed from the repo or if it's something else.

**Comment by andrii-bodnar** [comment]
Hi @CraigSiemens, the `--delete-obsolete` option is exactly what will help you in this case. 

This option tells CLI to delete files from the Crowdin project that previously matched the source pattern but now these files do not longer exist locally.

**Reply by CraigSiemens** [reply]
I've recently tried sending that option but it doesn't seem to remove files that are on longer in our repo. When the step runs it logs the following at the very end

```
✔️  No obsolete files were found
✔️  No obsolete directories found
```

But when we pull the translations later we still get the following warning.

```
Downloaded translations don't match the current project configuration. The translations for the following sources will be omitted (use --verbose to get the list of the omitted translations):
```

Followed by a list of files that are no longer present in the repo.

**Reply by anbraten** [reply]
@CraigSiemens This probably got fixed now by https://github.com/crowdin/crowdin-cli/pull/794 and https://github.com/crowdin/crowdin-cli/pull/776
